src/Microsoft.SymbolStore/KeyGenerators/ELFFileKeyGenerator.cs

```diff
@@ -108,16 +116,16 @@ public static IEnumerable<SymbolStoreKey> GetKeys(KeyTypeFlags flags, string pat
                 {
                     if (string.IsNullOrEmpty(symbolFileName))
                     {
-                        symbolFileName = path + SymbolFileExtensions[0];
+                        symbolFileName = path + s_symbolFileExtensions[0];
                     }
                     yield return BuildKey(symbolFileName, SymbolPrefix, buildId, "_.debug");
                 }
-                if ((flags & KeyTypeFlags.ClrKeys) != 0)
+                if ((flags & (KeyTypeFlags.ClrKeys | KeyTypeFlags.DacDbiKeys)) != 0)
                 {
                     /// Creates all the special CLR keys if the path is the coreclr module for this platform
                     if (GetFileName(path) == CoreClrFileName)
                     {
-                        foreach (string specialFileName in s_coreClrSpecialFiles)
+                        foreach (string specialFileName in (flags & KeyTypeFlags.DacDbiKeys) != 0 ? s_dacdbiSpecialFiles : s_coreClrSpecialFiles)
```

Comment:
If both `KeyTypeFlags.ClrKeys` & `KeyTypeFlags.DacDbiKeys` are set (maybe not an expected/supported condition) this will do the opposite of what would be wanted.
```suggestion
                        foreach (string specialFileName in (flags &KeyTypeFlags.ClrKeys) == 0 ? s_dacdbiSpecialFiles : s_coreClrSpecialFiles)
```

Also in a few other places...

Reply:
I reversed the checks so if both are set ClrKeys take precedence and returns all the special keys.